src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs

```diff
@@ -0,0 +1,352 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.CodeStyle;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Data;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Text;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Editor.EditorConfigSettings.Updater
+{
+    internal static partial class SettingsUpdateHelper
```

Comment:
@jmarolf This file should not have been added. It duplicates a lot of functionality from [ConfigurationUpdater](http://sourceroslyn.io/#Microsoft.CodeAnalysis.Features/CodeFixes/Configuration/ConfigurationUpdater.cs,9c4bc3b03dd44672) used in `Configure Severity` lightbulb action, and leads to many inconsistencies between how editorconfig is updated under the covers from Editorconfig UX versus `Configure Severity` lightbulb action (https://github.com/dotnet/roslyn/issues/52720). Latter also handles updating `category` based severities, which I believe is a feature request for Editorconfig UX.

Reply:
let me file a bug to explicitly track this. The intention was that I would send a cleanup PR to unify these implementations. As an aside: are there any tests for `ConfigurationUpdater`? I couldn't find any.

Reply:
https://github.com/dotnet/roslyn/issues/52735

Reply:
@jmarolf Not *directly* tested, but the tests in [here (VB)](https://github.com/dotnet/roslyn/tree/main/src/EditorFeatures/VisualBasicTest/Diagnostics/Configuration/ConfigureSeverity) and [here (C#)](https://github.com/dotnet/roslyn/tree/main/src/EditorFeatures/CSharpTest/Diagnostics/Configuration/ConfigureSeverity) test it indirectly (they test the `CodeFixProvider` which depends on `ConfigurationUpdated`)

Reply:
ah, thank you